includes/modules/category_row.php

```diff
@@ -14,40 +14,92 @@
-    if ($num_categories < MAX_DISPLAY_CATEGORIES_PER_ROW || MAX_DISPLAY_CATEGORIES_PER_ROW === 0) {
-        $col_width = floor(100/$num_categories);
-    } else {
-        $col_width = floor(100/MAX_DISPLAY_CATEGORIES_PER_ROW);
+$title = '';
+
+if (empty($num_categories)) {
+    return;
+}
+
+$columns_per_row = defined('MAX_DISPLAY_CATEGORIES_PER_ROW') ? (int)MAX_DISPLAY_CATEGORIES_PER_ROW : 0;
+$product_listing_layout_style = $columns_per_row > 1 ? 'columns' : 'fluid';
+
+// if in fixed-columns mode, calculate column width
+if ($product_listing_layout_style === 'columns') {
+    $calc_value = $columns_per_row;
+    if ($num_categories < $columns_per_row || $columns_per_row == 0) {
+        $calc_value = $num_categories;
     }
+    $col_width = floor(100 / $calc_value) - 0.5;
+}
 
-    foreach ($categories as $next_category) {
-        $zco_notifier->notify('NOTIFY_CATEGORY_ROW_IMAGE', $next_category['categories_id'], $next_category['categories_image']); 
-        if (empty($next_category['categories_image'])) {
-            $next_category['categories_image'] = 'pixel_trans.gif';
-        }
-        $cPath_new = zen_get_path($next_category['categories_id']);
+foreach ($categories as $next_category) {
+    $zco_notifier->notify('NOTIFY_CATEGORY_ROW_IMAGE', $next_category['categories_id'], $next_category['categories_image']);
+    if (empty($next_category['categories_image'])) {
+        $next_category['categories_image'] = 'pixel_trans.gif';
+    }
+    $cPath_new = zen_get_path($next_category['categories_id']);
 
-        // strip out 0_ from top level cats
-        $cPath_new = str_replace('=0_', '=', $cPath_new);
+    // strip out 0_ from top level cats
+    $cPath_new = str_replace('=' . (int)TOPMOST_CATEGORY_PARENT_ID . '_', '=', $cPath_new);
 
-        //    $categories->fields['products_name'] = zen_get_products_name($categories->fields['products_id']);
+    //    $categories->fields['products_name'] = zen_get_products_name($categories->fields['products_id']);
 
-        $list_box_contents[$row][$col] = [
-            'params' => 'class="categoryListBoxContents"' . ' ' . 'style="width:' . $col_width . '%;"',
-            'text' => '<a href="' . zen_href_link(FILENAME_DEFAULT, $cPath_new) . '">' . zen_image(DIR_WS_IMAGES . $next_category['categories_image'], $next_category['categories_name'], SUBCATEGORY_IMAGE_WIDTH, SUBCATEGORY_IMAGE_HEIGHT, 'loading="lazy"') . '<br>' . $next_category['categories_name'] . '</a>'
-        ];
 
-        $col++;
-        if ($col >= MAX_DISPLAY_CATEGORIES_PER_ROW) {
-            $col = 0;
-            $row++;
+
+    // set css classes for "row" wrapper, to allow for fluid grouping of cells based on viewport
+    // these defaults are based on Bootstrap4, but can be customized to suit your own framework
+    if ($product_listing_layout_style === 'fluid') {
+        $grid_cards_classes = 'row-cols-1 row-cols-md-2 row-cols-lg-2 row-cols-xl-3';
+        if (!isset($grid_classes_matrix)) {
```

Comment:
@scottcwilson This check for `!isset($grid_classes_matrix)` is the same as what the product_listing module does, and is the same as in the bootstrap class.

So if yours is not being picked up, maybe there's an order-of-operations issue?

Reply:
With the BS4 template, the default setup has the $center_column_width calculated to 6 (12 - 3 - 3 = 6). So the `$grid_classes_matrix` having only the 8 option uncommented would mean it won't take effect.